build/Shared/HashCodeCombiner.cs

```diff
@@ -79,29 +72,80 @@ internal void AddSequence<T>(IEnumerable<T> sequence)
             }
         }
 
+        internal void AddSequence<T>(T[] array)
+        {
+            if (array != null)
+            {
+                foreach (var item in array)
+                {
+                    AddObject(item);
```

Comment:
Why the approach for sequence is different from the one for dictionary?
Would it be better with `EnsureInitialize()` once and `AddInt32(item.GetHashCode())` for every item?